src/main/java/eu/bbmri_eric/negotiator/service/NetworkServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.TransactionException;
+
+@Transactional
+@Service
+public class NetworkServiceImpl implements NetworkService {
+  @Autowired NetworkRepository networkRepository;
+  @Autowired ResourceRepository resourceRepository;
+  @Autowired PersonRepository personRepository;
+  @Autowired ModelMapper modelMapper;
+  @Autowired private NetworkCreateModelMapper networkCreateModelMapper;
+
+  @Override
+  public NetworkDTO findNetworkById(Long id) {
+    return modelMapper.map(
+        networkRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(id)),
+        NetworkDTO.class);
+  }
+
+  @Override
+  public Iterable<NetworkDTO> findAllNetworks(Pageable pageable) {
+    return networkRepository
+        .findAll(pageable)
+        .map(network -> modelMapper.map(network, NetworkDTO.class));
+  }
+
+  @Override
+  public void deleteNetworkById(Long id) {
+    Network network = getNetwork(id);
+    networkRepository.delete(network);
+  }
+
+  @Override
+  public void removeResourceFromNetwork(Long networkId, Long resourceId) {
+    Network network = getNetwork(networkId);
+    Resource resource = getResource(resourceId);
+    network.removeResource(resource);
+    networkRepository.save(network);
+  }
+
+  @Override
+  public void removeManagerFromNetwork(Long networkId, Long managerId) {
+    Network network = getNetwork(networkId);
+    Person manager = getManager(managerId);
+    network.removeManager(manager);
+    networkRepository.save(network);
+  }
+
+  @Override
+  public NetworkDTO updateNetwork(Long id, NetworkCreateDTO networkDTO)
+      throws EntityNotStorableException {
+    Network network = getNetwork(id);
+    network.setName(networkDTO.getName());
+    network.setUri(networkDTO.getUri());
+    network.setExternalId(networkDTO.getExternalId());
+    network.setContactEmail(networkDTO.getContactEmail());
+    networkRepository.save(network);
+    return modelMapper.map(network, NetworkDTO.class);
+  }
+
+  @Override
+  public NetworkDTO createNetwork(NetworkCreateDTO networkCreateDTO)
+      throws EntityNotStorableException {
+    Network network = networkCreateModelMapper.mapToEntity(networkCreateDTO);
+    try {
+      Network savedNetwork = networkRepository.saveAndFlush(network);
+      return modelMapper.map(savedNetwork, NetworkDTO.class);
+    } catch (DataAccessException dae) {
+      throw new EntityNotStorableException("Database error occurred while saving network");
+    } catch (IllegalArgumentException iae) {
+      throw new EntityNotStorableException("Invalid data provided");
+    } catch (ConstraintViolationException cve) {
+      throw new EntityNotStorableException("Constraint violation");
+    } catch (TransactionException te) {
+      throw new EntityNotStorableException("Transaction error");
+    } catch (Exception ex) {
+      throw new EntityNotStorableException("An unexpected error occurred");
+    }
```

Comment:
please handle these exceptions in the https://github.com/BBMRI-ERIC/negotiator/blob/5993c7b47e2ee48056e858aaeac1020e1f55ef9c/src/main/java/eu/bbmri_eric/negotiator/exceptions/NegotiatorExceptionHandler.java class. This is I think fine but it looks cluttered IMHO. because we would then need to do this for every single method that stores an entity,